web/src/settings_org.ts

```diff
@@ -1070,70 +1203,91 @@ export function build_page() {
+        function (this: HTMLSelectOneElement, _event: JQuery.Event) {
+            const message_retention_setting_dropdown_value = this.value;
+            settings_components.change_element_block_display_property(
+                "id_realm_message_retention_custom_input",
+                message_retention_setting_dropdown_value === "custom_period",
+            );
+        },
+    );
 
-    $("#id_realm_waiting_period_threshold").on("change", function () {
-        const waiting_period_threshold = this.value;
-        settings_components.change_element_block_display_property(
-            "id_realm_waiting_period_threshold_custom_input",
-            waiting_period_threshold === "custom_period",
-        );
-    });
+    $<HTMLSelectOneElement>("select:not([multiple])#id_realm_waiting_period_threshold").on(
+        "change",
+        function (this: HTMLSelectOneElement, _event: JQuery.Event) {
+            const waiting_period_threshold = this.value;
+            settings_components.change_element_block_display_property(
+                "id_realm_waiting_period_threshold_custom_input",
+                waiting_period_threshold === "custom_period",
+            );
+        },
+    );
 
     $("#id_realm_digest_emails_enabled").on("change", function () {
         const digest_emails_enabled = $(this).is(":checked");
         settings_components.change_element_block_display_property(
             "id_realm_digest_weekday",
-            digest_emails_enabled === true,
+            digest_emails_enabled,
         );
     });
 
-    $("#id_realm_org_join_restrictions").on("change", function () {
-        const org_join_restrictions = this.value;
-        const $node = $("#allowed_domains_label").parent();
-        if (org_join_restrictions === "only_selected_domain") {
-            $node.show();
-            if (realm.realm_domains.length === 0) {
-                settings_realm_domains.show_realm_domains_modal();
+    $<HTMLSelectOneElement>("select:not([multiple])#id_realm_org_join_restrictions").on(
+        "change",
+        function (this: HTMLSelectOneElement, _event: JQuery.Event) {
+            const org_join_restrictions = this.value;
+            const $node = $("#allowed_domains_label").parent();
+            if (org_join_restrictions === "only_selected_domain") {
+                $node.show();
+                if (realm.realm_domains.length === 0) {
+                    settings_realm_domains.show_realm_domains_modal();
+                }
+            } else {
+                $node.hide();
             }
-        } else {
-            $node.hide();
-        }
-    });
+        },
+    );
 
-    $("#id_realm_allow_message_editing").on("change", function () {
-        const is_checked = $(this).prop("checked");
+    $("#id_realm_allow_message_editing").on("change", function (this: HTMLElement) {
+        const is_checked = Boolean($(this).prop("checked"));
         update_message_edit_sub_settings(is_checked);
     });
```

Comment:
```suggestion
    $<HTMLInputElement>("input#id_realm_allow_message_editing").on("change", function () {
        update_message_edit_sub_settings(this.checked);
    });
```

Reply:
Oh sweet I didn't know you could leave `this` out of the function args still! I'll update the other functions to do that, which will prevent some of the indentation changes.